src/others/file_explorer/lv_file_explorer.c

```diff
@@ -644,17 +669,30 @@ static void strip_ext(char * dir)
 static void exch_table_item(lv_obj_t * tb, int16_t i, int16_t j)
 {
     if(i == j) return;
+    const char * i_value = lv_table_get_cell_value(tb, i, 0);
+
+    char * tmp_i_value = lv_malloc(lv_strlen(i_value) + 1);
+    LV_ASSERT_MALLOC(tmp_i_value);
+    lv_strcpy(tmp_i_value, i_value);
+
+    //*Will be freed when the table is freed. The previous user data is freed when the new one is set*/
+    lv_file_explorer_file_table_entry_data_t * tmp_i_user_data = lv_malloc(sizeof(
```

Comment:
I couldn't find where it's freed. Is it?

Reply:
Inside `lv_table.c`:

The old data is freed when setting new data by using `lv_table_set_cell_user_data`:
```c
if(table->cell_data[cell]->user_data) {
    lv_free(table->cell_data[cell]->user_data);
}
```

And the cell user data is freed by the table destructor

Reply:
Oh I see. I missed it because I wasn't expecting the table widget to have the responsibility of calling `lv_free` on a user data field :sweat_smile: